ocaml/xcp-rrdd/bin/rrdp-netdev/rrdp_netdev.ml

```diff
@@ -18,6 +18,128 @@ module D = Debug.Make (struct let name = "xcp-rrdp-netdev" end)
 
 module Process = Rrdd_plugin.Process (struct let name = "xcp-rrdd-netdev" end)
 
+type iface_stats = {
+    tx_bytes: int64  (** bytes emitted *)
+  ; tx_pkts: int64  (** packets emitted *)
+  ; tx_errors: int64  (** error emitted *)
+  ; rx_bytes: int64  (** bytes received *)
+  ; rx_pkts: int64  (** packets received *)
+  ; rx_errors: int64  (** error received *)
+}
+
+let default_stats =
+  {
+    tx_bytes= 0L
+  ; tx_pkts= 0L
+  ; tx_errors= 0L
+  ; rx_bytes= 0L
+  ; rx_pkts= 0L
+  ; rx_errors= 0L
+  }
+
+let monitor_whitelist =
+  ref
+    [
+      "eth"
+    ; "vif" (* This includes "tap" owing to the use of standardise_name below *)
+    ]
+
+let standardise_name name =
+  try
+    let d1, d2 = Scanf.sscanf name "tap%d.%d" (fun d1 d2 -> (d1, d2)) in
+    let newname = Printf.sprintf "vif%d.%d" d1 d2 in
+    newname
+  with _ -> name
+
+let get_link_stats () =
+  let open Netlink in
+  let s = Socket.alloc () in
+  Socket.connect s Socket.NETLINK_ROUTE ;
+  let cache = Link.cache_alloc s in
+  let links = Link.cache_to_list cache in
+  let links =
+    let is_whitelisted name =
+      List.exists
+        (fun s -> Astring.String.is_prefix ~affix:s name)
+        !monitor_whitelist
```

Comment:
I assume this list is short. Otherwise it would be better to use a set or hashtbl.

Reply:
yes, it's defined just above with two entries currently. it was extendable at startup in networkd, but it was empty by default and rrd output can be filtered afterwards with ease, so it's static here.